app/models/provider.rb

```diff
@@ -4,4 +4,40 @@ class Provider < ApplicationRecord
   has_and_belongs_to_many :organisations, join_table: :organisation_provider
 
   has_many :sites
+  has_many :enrichments, foreign_key: :provider_code, primary_key: :provider_code, class_name: "ProviderEnrichment"
+
+  def address
+    if enrichments.present?
+      attrs = enrichments.last.json_data
+      {
+        "address1" => attrs["Address1"],
+        "address2" => attrs["Address2"],
+        "address3" => attrs["Address3"],
+        "address4" => attrs["Address4"],
+        "postcode" => attrs["Postcode"]
+      }
+    else
+      attributes
```

Comment:
This returns more than just the address, no?

Reply:
It does. It's not the tidiest, true, but the whole ordeal is a bit messy anyway. I'd be up for tidying it up in the future, I reckon we'll be revisiting this bit of code anyway.